### PR TITLE
autopsy: add integration for d application

### PR DESCRIPTION
**Pull request by ljmf00**
Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

File: gboard_autopsy.py [hunk 1]
```diff
@@ -121,3 +142,130 @@ def createCustomArtifactType(self, current_case, art_type_name, art_desc):
...
+        Returns:
+            A BlackboardAttribute.Type instance created by the sleuthkit case.
+        """
+
+        sk_case = current_case.getSleuthkitCase()
+        try:
+            sk_case.addArtifactAttributeType(attr_type_name, BlackboardAttribute.TSK_BLACKBOARD_ATTRIBUTE_VALUE_TYPE.STRING, attr_desc)
+        except (TskCoreException, TskDataException) as ex:
+            self.log(Level.INFO, 'Error creating attribute type: ' + attr_type_name)
+            self.log(Level.INFO, ex.toString())
```
Comment (iK4tsu) [inline gboard_autopsy.py]: Change this to `SEVERE`

File: gboard_autopsy.py [hunk 2]
```diff
@@ -121,3 +142,130 @@ def createCustomArtifactType(self, current_case, art_type_name, art_desc):
...
+    def analyze_clipboard(self, blackboard, data_source, file_manager):
+        """Analyze gboard clipboard content and publish to blackboard
+
+        Args:
+            blackboard (Blackboard): blackboard artifact indexer
+            data_source (Content): autopsy case data source
+            file_manager (FileManager): file manager service
+        """
+
+        files = file_manager.findFiles(data_source, '%', self.GBOARD_PACKAGE_NAME + '/files/clipboard_image')
```
Comment (iK4tsu) [inline gboard_autopsy.py]: Add an additional `/` at the end

File: gboard_autopsy.py [hunk 3]
```diff
@@ -121,3 +142,130 @@ def createCustomArtifactType(self, current_case, art_type_name, art_desc):
...
+        return [
+            self.path_to_exe,
+            '-d', input_dir,
+        ]
+
+    def get_input_dir(self, data_source, file_manager):
+        files = file_manager.findFiles(data_source, "%" + self.GBOARD_PACKAGE_NAME + "%")
+
+        for file in files:
+            if file.isFile() == True:
```
Comment (iK4tsu) [inline gboard_autopsy.py]: Can't we remove `== True`?

File: gboard_autopsy.py [hunk 4]
```diff
@@ -121,3 +142,130 @@ def createCustomArtifactType(self, current_case, art_type_name, art_desc):
...
+                path = file.getLocalAbsPath()
+                if path == None:
```
Comment (iK4tsu) [inline gboard_autopsy.py]: ditto for `== None`

File: gboard_autopsy.py [hunk 5]
```diff
@@ -121,3 +142,130 @@ def createCustomArtifactType(self, current_case, art_type_name, art_desc):
...
+        if not common_path.endswith(os.path.sep):
+            sanitized_path = common_path + os.path.sep
+        else:
+            sanitized_path = common_path
+
+        rel_path = full_path.split(sanitized_path)
+        if len(rel_path) > 1:
+            full_gboard_path = os.path.join(self.GBOARD_PACKAGE_NAME, os.path.dirname(rel_path[1]))
+            files = file_manager.findFiles(data_source, os.path.basename(rel_path[1]), full_gboard_path)
+            return files[0] if len(files) else None
```
Comment (iK4tsu) [inline gboard_autopsy.py]: You can replace `len(files)` with `files`

File: gboard_autopsy.py [hunk 6]
```diff
@@ -121,3 +142,130 @@ def createCustomArtifactType(self, current_case, art_type_name, art_desc):
...
+
+    def publish_analysis_artifact(self, blackboard, file, artifact_type, attr_list = []):
+        artifact = file.newArtifact(artifact_type.getTypeID())
+
+        blackboard_attr_list = []
+        for attr_type, attr_content in attr_list:
+            blackboard_attr_list.append(BlackboardAttribute(attr_type,
+            GboardDataSourceIngestModuleFactory.moduleName, attr_content))
+
+        if len(blackboard_attr_list) > 0:
```
Comment (iK4tsu) [inline gboard_autopsy.py]: ditto

**Review by iK4tsu — CHANGES_REQUESTED** [review]

File: gboard_autopsy.py (hunk 1 above)
Comment (ljmf00) [reply gboard_autopsy.py]: Done

File: gboard_autopsy.py (hunk 2 above)
Comment (ljmf00) [reply gboard_autopsy.py]: Done

File: gboard_autopsy.py (hunk 3 above)
Comment (ljmf00) [reply gboard_autopsy.py]: Done

File: gboard_autopsy.py (hunk 4 above)
Comment (ljmf00) [reply gboard_autopsy.py]: Done

File: gboard_autopsy.py (hunk 5 above)
Comment (ljmf00) [reply gboard_autopsy.py]: Done

File: gboard_autopsy.py (hunk 6 above)
Comment (ljmf00) [reply gboard_autopsy.py]: Done

**Review by iK4tsu — APPROVED** [review]

**Pull request merged**